### PR TITLE
Sub host options: ala whitecolor

### PR DESCRIPTION
**Pull request by gojisystems**
Taken from @whitecolor 's work on Express version of server. Applied to new server, including working tests (also taken from @whitecolor). Main modification occurred to GetClientIdFromHostname function.

### CODE REVIEW

**Comment by gojisystems** [comment]
fixed travis build bugs/tests. now allow for single string sub-hosts. 
`--sub-host nested.sub` works as well as 
`--sub-host nested --sub-host sub`

**Comment by defunctzombie** [comment]
What is a "sub-host"? Would it be easier if the subdomain parsing just looked at items before the first period? Or are there other features preventing that?

**Comment by defunctzombie** [comment]
Specifically; how do these approaches compare to https://github.com/localtunnel/server/pull/25

**Comment by defunctzombie** [comment]
And this one https://github.com/localtunnel/server/pull/24

File: README.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
 # localtunnel-server
 
-[![Build Status](https://travis-ci.org/localtunnel/server.svg?branch=master)](https://travis-ci.org/localtunnel/server)
+[![Build Status](https://travis-ci.org/gojisystems/gojiconnect-server.svg?branch=master)](https://travis-ci.org/gojisystems/gojiconnect-server)
```
Comment (defunctzombie) [inline README.md]: When making a PR please don't alter files in this way as such a change can't be merged into this repo as-is. This is great for your repo but for the PR it should be removed.

**Comment by defunctzombie** [comment]
Done via 86dce49dd1d40a64a2f121ff3aa271719417810f

Thank you for your input on this feature!

**Pull request closed without merging**